From 1d078f2fd9f97ce947f432dc40ae7d4b24e5fdbc Mon Sep 17 00:00:00 2001
From: Connor <iconnoroneill@gmail.com>
Date: Tue, 20 Jun 2017 15:13:31 +0930
Subject: [PATCH 06/17] begin creating frmAddModifyState

Signed-off-by: Connor <iconnoroneill@gmail.com>
---
 Student Files/VendorMaintenance/VendorMaintenance.csproj | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Student Files/VendorMaintenance/VendorMaintenance.csproj b/Student Files/VendorMaintenance/VendorMaintenance.csproj
index f9d7590..e480f3e 100644
--- a/Student Files/VendorMaintenance/VendorMaintenance.csproj	
+++ b/Student Files/VendorMaintenance/VendorMaintenance.csproj	
@@ -61,6 +61,12 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="DataContext.cs" />
+    <Compile Include="frmAddModifyState.cs">
+      <SubType>Form</SubType>
+    </Compile>
+    <Compile Include="frmAddModifyState.Designer.cs">
+      <DependentUpon>frmAddModifyState.cs</DependentUpon>
+    </Compile>
     <Compile Include="frmAddModifyVendor.cs">
       <SubType>Form</SubType>
     </Compile>
-- 
2.11.0.windows.1

